js/engine.js

```diff
@@ -814,7 +814,20 @@ class Engine {
     const currentValueRaw = self.getStream(stream);
     const currentValue = unitConverter.convert(currentValueRaw, "kg");
 
-    const convertedMax = unitConverter.convert(amount, "kg");
+    let convertedMax;
+
+    // Check if amount is specified in equipment units and add recharge if needed
+    if (amount.getUnits().startsWith("unit")) {
```

Comment:
Please add `hasEquipmentUnits()` to `EngineNumber` that has this logic and update to use that method call where this logic is repeated.